src/app.rs

```diff
@@ -0,0 +1,138 @@
+use crate::cli::{Cli, Command};
+use crate::cloud_client::CloudClient;
+use crate::errors::AppError;
+use crate::tui::WorkMode;
+use crate::utilities::files::get_path_entries;
+use std::path::PathBuf;
+use tracing::debug;
+
+pub struct WorkspaceData {
+    pub local_path: PathBuf,
+    pub local_entries: Vec<String>,
+    pub cloud_path: PathBuf,
+    pub cloud_entries: Vec<String>,
+}
+
+impl Default for WorkspaceData {
+    fn default() -> Self {
+        Self {
+            local_path: std::env::current_dir().unwrap_or_default(),
+            local_entries: vec![],
+            cloud_path: PathBuf::from("//"),
+            cloud_entries: vec![],
+        }
+    }
+}
+
+pub struct App<C: CloudClient> {
+    pub input_command: String,
+    pub cursor_position: usize,
```

Comment:
Do you need all fields to be public?

Reply:
Yes, those fields are used at `ui` render function